In [2]:
#Import library 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#data importing

In [3]:
CM=pd.read_excel('E-commerce.xlsx')
CM

InvoiceNo StockCode                          Description  Quantity  \
0       536365.0    85123A   WHITE HANGING HEART T-LIGHT HOLDER       6.0   
1       536365.0   71053.0                  WHITE METAL LANTERN       6.0   
2       536365.0    84406B       CREAM CUPID HEARTS COAT HANGER       8.0   
3       536365.0    84029G  KNITTED UNION FLAG HOT WATER BOTTLE       6.0   
4       536365.0    84029E       RED WOOLLY HOTTIE WHITE HEART.       6.0   
...          ...       ...                                  ...       ...   
230584  581587.0   22613.0          PACK OF 20 SPACEBOY NAPKINS      12.0   
230585  581587.0   22899.0          CHILDREN'S APRON DOLLY GIRL       6.0   
230586  581587.0   23254.0         CHILDRENS CUTLERY DOLLY GIRL       4.0   
230587  581587.0   23255.0      CHILDRENS CUTLERY CIRCUS PARADE       4.0   
230588  581587.0   22138.0         BAKING SET 9 PIECE RETROSPOT       3.0   

               InvoiceDate  UnitPrice  CustomerID  Revenue         Country  
0      2010-12-01 08:26:00       2.55     17850.0    15.30  United Kingdom  
1      2010-12-01 08:26:00       3.39     17850.0    20.34  United Kingdom  
2      2010-12-01 08:26:00       2.75     17850.0    22.00  United Kingdom  
3      2010-12-01 08:26:00       3.39     17850.0    20.34  United Kingdom  
4      2010-12-01 08:26:00       3.39     17850.0    20.34  United Kingdom  
...                    ...        ...         ...      ...             ...  
230584 2011-12-09 12:50:00       0.85     12680.0    10.20          France  
230585 2011-12-09 12:50:00       2.10     12680.0    12.60          France  
230586 2011-12-09 12:50:00       4.15     12680.0    16.60          France  
230587 2011-12-09 12:50:00       4.15     12680.0    16.60          France  
230588 2011-12-09 12:50:00       4.95     12680.0    14.85          France  

[230589 rows x 9 columns]

In [4]:
# There are rows and columns in the data
CM.shape

(230589, 9)

In [5]:
# code for finding data type and null value checking
CM.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 230589 entries, 0 to 230588
Data columns (total 9 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   InvoiceNo    230589 non-null  object        
 1   StockCode    230589 non-null  object        
 2   Description  229991 non-null  object        
 3   Quantity     230589 non-null  float64       
 4   InvoiceDate  230589 non-null  datetime64[ns]
 5   UnitPrice    230589 non-null  float64       
 6   CustomerID   170491 non-null  float64       
 7   Revenue      230589 non-null  float64       
 8   Country      230589 non-null  object        
dtypes: datetime64[ns](1), float64(4), object(4)
memory usage: 15.8+ MB


### 1. Find the total revenue 


In [6]:
a=CM['Revenue'].sum()
a


5148073.239999999

### 2. Find the total number of Orders


In [7]:
b=CM['InvoiceNo'].count()
b

230589

### 3. Find the unique number of customers, and give which customer most frequently using e commerce sites.


In [15]:
B=CM['CustomerID'].value_counts()
B


17841.0    3368
14911.0    2313
12748.0    2078
14096.0    1604
14606.0    1236
           ... 
13270.0       1
17736.0       1
14682.0       1
15724.0       1
18080.0       1
Name: CustomerID, Length: 3125, dtype: int64

### 4. Find the revenue by country


In [10]:
s=CM.groupby('Country').sum()['Revenue']
s

Country
Australia                 51918.01
Austria                    2999.69
Bahrain                     665.14
Belgium                   17086.58
Canada                     1751.88
Channel Islands            7899.21
Cyprus                     5673.51
Czech Republic              334.99
Denmark                    4998.11
EIRE                     113425.86
European Community          440.55
Finland                   11519.79
France                    87804.11
Germany                   97755.23
Greece                     2149.28
Hong Kong                  1550.59
Iceland                    2540.29
Israel                     1045.75
Italy                      3797.76
Japan                     19074.65
Lithuania                  1661.06
Malta                       863.97
Netherlands               89945.76
Norway                    15495.40
Poland                     3983.12
Portugal                  13776.38
Saudi Arabia                 14.75
Singapore                 14300.42
Spain       

### 5. Find the Total number products sold


In [11]:
b=CM['Quantity'].sum()
b

2638512.0

### 6. Find the top 3 countries in terms of revenue


In [12]:
CM.groupby('Country').sum()["Revenue"].nlargest(3)

Country
United Kingdom    4502149.94
EIRE               113425.86
Germany             97755.23
Name: Revenue, dtype: float64

### 7. Find the average revenue by country


In [13]:
s=CM.groupby('Country').mean()['Revenue']
s

Country
Australia               132.782634
Austria                  27.774907
Bahrain                  47.510000
Belgium                  19.662348
Canada                   22.460000
Channel Islands          21.760909
Cyprus                   23.838277
Czech Republic           27.915833
Denmark                  36.750809
EIRE                     36.065456
European Community       20.978571
Finland                  31.734959
France                   21.776813
Germany                  24.698138
Greece                   18.853333
Hong Kong                46.987576
Iceland                  23.965000
Israel                   13.407051
Italy                    22.605714
Japan                    87.098858
Lithuania                47.458857
Malta                    17.632041
Netherlands             110.770640
Norway                   32.829237
Poland                   20.637927
Portugal                 18.271061
Saudi Arabia             14.750000
Singapore               230.651935
Spain       

### 8. Find the Maximum revenue by country

In [14]:
s=CM.groupby('Country').max()['Revenue']
s

C:\Users\user\AppData\Local\Temp\ipykernel_9768\3722682840.py:1: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  s=CM.groupby('Country').max()['Revenue']


Country
Australia                 1944.00
Austria                    160.00
Bahrain                    205.74
Belgium                    165.00
Canada                     534.24
Channel Islands            131.40
Cyprus                     290.00
Czech Republic              61.20
Denmark                    266.24
EIRE                      1687.17
European Community          45.00
Finland                    551.20
France                    1136.30
Germany                    876.00
Greece                     135.00
Hong Kong                  166.00
Iceland                    249.60
Israel                      75.00
Italy                      120.00
Japan                     3794.40
Lithuania                  122.40
Malta                      160.00
Netherlands               2577.60
Norway                     376.50
Poland                     102.00
Portugal                   557.72
Saudi Arabia                14.75
Singapore                 2382.92
Spain                     1350.00
Sweden

# Conclusion



1.This is a E - Commerce.

2)The data contains 230589 rows and 9 columns.

3)There is No null values in this data.

4) Total revenue is 5148073.239999999.

5) Total number of Orders is 230589.

6)The revenue by country (Canada)-1751.88.

7)Total number products sold-230589

8)Top 3 countries in terms of revenue

9)Average revenue by country(EIRE)-36.065456

10)Maximum revenue by country (United Kingdom)-168469.60